### PR TITLE
Fix missing validation of mutable samplers

### PR DESCRIPTION
**Pull request by hakzsam**

### CODE REVIEW

File: cli/fossilize_feature_filter.cpp [hunk 1]
```diff
@@ -1000,6 +1003,11 @@ bool FeatureFilter::Impl::descriptor_set_layout_is_supported(const VkDescriptorS
 					mutable_counts.storage_image = 1;
 					break;
 
+				case VK_DESCRIPTOR_TYPE_SAMPLER:
+					// Sampler is a non-standard type for mutable
+					must_check_set_layout_before_accept = true;
```
Comment (HansKristian-Work) [inline cli/fossilize_feature_filter.cpp:1008]: Should also set the mutable counts I think. It just also needs to set this bool to true.

File: cli/device.cpp [hunk 2]
```diff
@@ -518,6 +518,17 @@ bool VulkanDevice::format_is_supported(VkFormat format, VkFormatFeatureFlags for
...
 }
 
+bool VulkanDevice::descriptor_set_layout_is_supported(const VkDescriptorSetLayoutCreateInfo *info)
+{
+	if (is_null_device)
+		return true;
+
+	VkDescriptorSetLayoutSupport set_layout_support = { VK_STRUCTURE_TYPE_DESCRIPTOR_SET_LAYOUT_SUPPORT };
+
+	vkGetDescriptorSetLayoutSupport(device, info, &set_layout_support);
```
Comment (HansKristian-Work) [inline cli/device.cpp:528]: Probably a little more safe to use the KHR one in case a 1.0 device is used. maintenance3 is required for mutable, not 1.1.

Comment (hakzsam) [reply cli/device.cpp]: Sounds good.

File: cli/fossilize_feature_filter.cpp (hunk 1 above)
Comment (hakzsam) [reply cli/fossilize_feature_filter.cpp:1009]: Should be fixed.

**Review by HansKristian-Work — APPROVED** [review]
LGTM

**Pull request merged**